Makefile

```
@@ -80,6 +80,8 @@ environment:
 			$$PYTHON_VENV_PATH/bin/python -m pip install --extra-index-url https://test.pypi.org/simple/ PennyLane-Lightning --pre --upgrade;\
 			$$PYTHON_VENV_PATH/bin/python -m pip install --upgrade git+https://github.com/PennyLaneAI/pennylane.git#egg=pennylane;\
 			$$PYTHON_VENV_PATH/bin/python -m pip install --upgrade git+https://github.com/XanaduAI/iqpopt.git#egg=iqpopt;\
+			# TODO: Remove this on next release. Currently Catalyst 0.11 pins jaxlib to 0.4.8;\
+			$$PYTHON_VENV_PATH/bin/python -m pip install 'numpy<2';\
```

I'm pretty sure Catalyst supports Numpy 2, it just can't be `2.0.0`. Is that related to jaxlib though?
```suggestion
			# TODO: Remove this on next release. Currently Catalyst 0.11 pins jaxlib to 0.4.28;\
			$$PYTHON_VENV_PATH/bin/python -m pip install 'numpy<2';\
```

---

Hey @dime10 , it's moreof a pip issue than Catalyst.

When it upgrades jax pip also upgrades all pre-installed packages to their latest version.


I think a better TODO would be

```python
# TODO: Remove this on next release once QML is upgraded to JAX > 0.4.28
```

---

Since there is no code in this demo I rebased to `master`. Should I remove this or leave as is?

---

It's passing in the PR, so fine for me atm, just dont want to merge it to master if its not necessary there

---

dev didnt build due to some dependency conflicts

---

I _think_ `dev` was pinned to numpy<2 for some other reason, that was the main problem afaik

[slack context](https://xanaduhq.slack.com/archives/C03FRM3RCCV/p1746776486825339)

---

Ok so from the slack context it sounds like this statement is true:
>Are you saying the qml repo currently doesn't work numpy 2 then?

And the problem is that installing a different version of JAX seems to install  a different version of NumPy? 

Anyway I'll let @rashidnhm answer/close off this thread then, since he seems to have the most context.